go/geth_adapter/adapter_test.go

```diff
@@ -75,3 +82,400 @@ func TestRunContextAdapter_SetBalanceHasCorrectEffect(t *testing.T) {
+	stateDb.EXPECT().SetCode(gc.Address(address), code)
+	created := adapter.CreateAccount(address, code)
+	if !created {
+		t.Errorf("Account should have been created")
+	}
+
+	stateDb.EXPECT().AddressInAccessList(gc.Address(address)).Return(true)
+	inAccessList := adapter.IsAddressInAccessList(address)
+	if !inAccessList {
+		t.Errorf("Address should be in access list")
+	}
+}
+
+type testContractRef struct {
+	address gc.Address
+}
+
+func (c testContractRef) Address() gc.Address {
+	return c.address
+}
+
+func TestRunContextAdapter_Call(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+
+	address := gc.Address{0x42}
+
+	stateDb.EXPECT().Snapshot().Return(1)
+	stateDb.EXPECT().Exist(address).Return(true)
+	stateDb.EXPECT().GetCode(address).Return([]byte{})
+	stateDb.EXPECT().Witness()
+
+	contractRef := testContractRef{address: address}
+
+	canTransfer := func(geth.StateDB, gc.Address, *uint256.Int) bool { return true }
+	transfer := func(geth.StateDB, gc.Address, gc.Address, *uint256.Int) {}
+
+	runContextAdapter := &runContextAdapter{
+		evm: &geth.EVM{
+			StateDB: stateDb,
+			Context: geth.BlockContext{
+				CanTransfer: canTransfer,
+				Transfer:    transfer,
+			},
+		},
+		contract: geth.NewContract(contractRef, contractRef, nil, 0),
+	}
+
+	gas := tosca.Gas(42)
+
+	parameters := tosca.CallParameters{
+		Recipient: tosca.Address(address),
+		Gas:       gas,
+	}
+
+	result, err := runContextAdapter.Call(tosca.Call, parameters)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if !result.Success {
+		t.Errorf("Call was not successful")
+	}
+	if result.GasLeft != gas {
+		t.Errorf("Call has the wrong amount of gas left: %v, expected: %v", result.GasLeft, gas)
+	}
+}
+
+func TestRunContextAdapter_Run(t *testing.T) {
+	tests := map[string]bool{
+		"success": true,
+		"failure": false,
+	}
+
+	for name, success := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			stateDb := NewMockStateDb(ctrl)
+			interpreter := tosca.NewMockInterpreter(ctrl)
+
+			chainId := int64(42)
+			blockNumber := int64(24)
+			address := tosca.Address{0x42}
+
+			blockParameters := geth.BlockContext{BlockNumber: big.NewInt(blockNumber)}
+			chainConfig := &params.ChainConfig{ChainID: big.NewInt(chainId)}
+			evm := geth.NewEVM(blockParameters, geth.TxContext{}, stateDb, chainConfig, geth.Config{})
+			adapter := &gethInterpreterAdapter{
+				evm:         evm,
+				interpreter: interpreter,
+			}
+
+			blockParams := tosca.BlockParameters{
+				ChainID:     tosca.Word(tosca.NewValue(uint64(chainId))),
+				BlockNumber: blockNumber,
+			}
+			expectedParams := tosca.Parameters{
+				BlockParameters: blockParams,
+				Kind:            tosca.Call,
+				Static:          false,
+				Recipient:       address,
+				Sender:          address,
+			}
+
+			interpreter.EXPECT().Run(parametersMatcher{expectedParams}).Return(tosca.Result{Success: success}, nil)
```

Comment:
fyi: since it is only used once, you could have implemented it using `DoAndReturn`, avoiding the need to implement a matcher:
```
interpreter.EXPECT().Run(gomock.Any()).DoAndReturn(func(params tosca.Parameters) (tosca.Result, error) {
    if params.Kind != tosca.Call {
	t.Errorf("Unexpected kind: %v", params.Kind)
    }
    if blockParams != params.BlockParameters {
	t.Errorf("Unexpected block parameters: %v", params.BlockParameters)
    }
    // ... all the other stuff ...
    return tosca.Result{Success: success}, nil
})
```